### PR TITLE
Fix:SPCTR-2551- Post carousel - Gap between post not visible when only 2 post is present.

### PR DESCRIPTION
**Pull request by ravinishadcoder**
### Description

Post carousel - Gap between post not visible when only 2 post is present.
[task link](https://brainstormforce.atlassian.net/browse/SPCTR-2551?atlOrigin=eyJpIjoiYmVkNmJlMmUyZjVkNGFmOTlkMTgzZjMxZDY1ZjM2YTQiLCJwIjoiaiJ9)
### Screenshots

[screencast](https://d.pr/v/j5zgYX)
### Types of changes

Bug fix (non-breaking change which fixes an issue) 

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: blocks-config/post/class-uagb-post.php [hunk 1]
```diff
@@ -1712,6 +1712,21 @@ public function add_post_dynamic_script() {
...
+								for( let article of articles ) {
+									let articleWidth = article.offsetWidth;
+									let rowGap = <?php echo esc_html( $value['rowGap'] ); ?>;
+									let imageWidth = 100 - ( rowGap / articleWidth ) * 100;
+									let image = article.getElementsByClassName('uagb-post__image');
+									if ( image[0] ) {
+										image[0].style.width = imageWidth + '%';
+										image[0].style.marginLeft = rowGap / 2 + 'px';
+
+									}
```
Comment (rahulvarma722) [inline blocks-config/post/class-uagb-post.php:1728]:
```suggestion
							let image = article.getElementsByClassName('uagb-post__image');
									if ( image[0] ) {
									let articleWidth = article.offsetWidth;
									let rowGap = <?php echo esc_html( $value['rowGap'] ); ?>;
									let imageWidth = 100 - ( rowGap / articleWidth ) * 100;

										image[0].style.width = imageWidth + '%';
										image[0].style.marginLeft = rowGap / 2 + 'px';

									}
```

**Review by rahulvarma722 — CHANGES_REQUESTED** [review]

File: blocks-config/post/class-uagb-post.php [hunk 2]
```diff
@@ -1712,6 +1712,22 @@ public function add_post_dynamic_script() {
 							( function( $ ) {
 								var cols = parseInt( '<?php echo esc_html( $value['columns'] ); ?>' );
 								var $scope = $( '.uagb-block-<?php echo esc_html( $key ); ?>' );
+
+								// This CSS is for Post BG Image Spacing
+								let articles = document.querySelectorAll( '.uagb-post__image-position-background .uagb-post__inner-wrap' );
+
+								for( let article of articles ) {
+							let image = article.getElementsByClassName('uagb-post__image');
```
Comment (rahulvarma722) [inline blocks-config/post/class-uagb-post.php:1720]: Add proper indentation.

File: blocks-config/post/class-uagb-post.php [hunk 3]
```diff
@@ -1712,6 +1712,22 @@ public function add_post_dynamic_script() {
...
+									if ( image[0] ) {
+									let articleWidth = article.offsetWidth;
```
Comment (rahulvarma722) [inline blocks-config/post/class-uagb-post.php:1722]: Add proper indentation.

**Review by rahulvarma722 — CHANGES_REQUESTED** [review]

File: blocks-config/post/class-uagb-post.php [hunk 4]
```diff
@@ -1712,6 +1712,21 @@ public function add_post_dynamic_script() {
 							( function( $ ) {
 								var cols = parseInt( '<?php echo esc_html( $value['columns'] ); ?>' );
 								var $scope = $( '.uagb-block-<?php echo esc_html( $key ); ?>' );
+
+								// This CSS is for Post BG Image Spacing
+								let articles = document.querySelectorAll( '.uagb-post__image-position-background .uagb-post__inner-wrap' );
+
+								for( let article of articles ) {
```
Comment (rahulvarma722) [inline blocks-config/post/class-uagb-post.php:1719]: Validate articles first.

**Review by rahulvarma722 — CHANGES_REQUESTED** [review]

**Comment by supriyajjadhav** [comment]
Please add changelog

File: blocks-config/post/class-uagb-post.php (hunk 4 above)
Comment (ravinishadcoder) [reply blocks-config/post/class-uagb-post.php:1722]: Added validation for articles

**Review by AbhijitNage123 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

File: package-lock.json
Comment (RajkiranBagal) [inline package-lock.json:1]: @ravinishadcoder Undo this change. Copy next-release file & paste it if needed.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (ravinishadcoder) [reply package-lock.json:1]: I have undo the changes

**Review by RajkiranBagal — APPROVED** [review]

File: blocks-config/post/class-uagb-post.php [hunk 5]
```diff
@@ -1712,6 +1712,24 @@ public function add_post_dynamic_script() {
 							( function( $ ) {
 								var cols = parseInt( '<?php echo esc_html( $value['columns'] ); ?>' );
 								var $scope = $( '.uagb-block-<?php echo esc_html( $key ); ?>' );
+
+								// This CSS is for Post BG Image Spacing
+								let articles = document.querySelectorAll( '.uagb-post__image-position-background .uagb-post__inner-wrap' );
+								if( articles === undefined) {
+									return;
+								}
```
Comment (rahulvarma722) [inline blocks-config/post/class-uagb-post.php:1720]:
```suggestion
								if( articles?.length ) {
									return;
								}
```
This should be

**Review by rahulvarma722 — CHANGES_REQUESTED** [review]

File: blocks-config/post/class-uagb-post.php (hunk 5 above)
Comment (ravinishadcoder) [reply blocks-config/post/class-uagb-post.php:1720]: I implemented the code you suggested. But it is not working as expected. This is always returned from line number 1720.
[Screencast](https://d.pr/v/4GRnco)

**Review by RajkiranBagal — APPROVED** [review]

**Review by rahulvarma722 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**